types/errors.go

```diff
@@ -31,8 +33,9 @@ const (
 	CodeUnknownAddress    CodeType = 9
 	CodeInsufficientCoins CodeType = 10
 	CodeInvalidCoins      CodeType = 11
+	CodeGeneric           CodeType = 12
```

Comment:
Clients can use the code (and only the code) to display a short internationalized error message
So, every type of error with its own formatting string should have its own code.
We can support a "CodeUnknown", but "CodeGeneric" sounds like people should be using that, but we don't want to encourage that.